packages/tools/kadena-cli/src/tx/utils/txHelpers.ts

```diff
@@ -0,0 +1,341 @@
+  } catch (error) {
+    throw new Error(`Error signing transaction: ${error.message}`);
+  }
+}
+
+export async function signTransactionWithKeyPair(
+  keys: IKeyPairLocal[],
+  unsignedTransactions: IUnsignedCommand[],
+  legacy?: boolean,
+): Promise<(ICommand | IUnsignedCommand | undefined)[]> {
+  try {
+    const signedTransactions: (ICommand | IUnsignedCommand | undefined)[] = [];
+
+    for (const unsignedCommand of unsignedTransactions) {
+      const parsedTransaction = JSON.parse(unsignedCommand.cmd);
+      const relevantKeyPairs = getRelevantKeypairs(parsedTransaction, keys);
+
+      if (relevantKeyPairs.length === 0) {
+        throw new Error(
+          'No matching signable keys found between wallet and transaction:',
+        );
+      }
+
+      if (legacy === true) {
+        const signatures = await Promise.all(
+          relevantKeyPairs.map(async (key) => {
+            const sigUint8Array = await legacyKadenaSign(
+              '',
+              unsignedCommand.cmd,
+              key.secretKey as EncryptedString,
+            );
+
+            const sig = Buffer.from(sigUint8Array).toString('hex');
+            return { sig, pubKey: key.publicKey };
+          }),
+        );
+
+        const command = addSignatures(unsignedCommand, ...signatures);
+        signedTransactions.push(command);
+      } else {
+        const signWithKeypair = createSignWithKeypair(
+          relevantKeyPairs as IKeyPair[],
+        );
+        const command = await signWithKeypair(unsignedCommand);
+        signedTransactions.push(command);
+      }
+    }
+    return signedTransactions;
+  } catch (error) {
+    throw new Error(`Error signing transaction: ${error.message}`);
+  }
+}
+
+export function getRelevantKeypairs(
+  tx: IPactCommand,
+  keypairs: IKeyPairLocal[],
+): IKeyPairLocal[] {
+  const relevantKeypairs = keypairs.filter((keypair) =>
+    tx.signers.some(({ pubKey }) => pubKey === keypair.publicKey),
+  );
+  return relevantKeypairs;
+}
+
+/**
+ * retrieve transaction from file
+ *
+ * @param {string} transactionFile
+ * @param {string} path
+ * @param {boolean} signed
+ * @returns {Promise<IUnsignedCommand | ICommand>}
+ * @throws Will throw an error if the file cannot be read or the transaction cannot be processed.
+ */
+export async function getTransactionFromFile(
+  transactionFile: string,
+  signed: boolean,
+  path?: string,
+): Promise<IUnsignedCommand | ICommand> {
+  try {
+    const filePath =
+      path !== undefined
+        ? join(process.cwd(), path, transactionFile)
+        : join(TRANSACTION_PATH, transactionFile);
+    const transactionFilePath = join(TRANSACTION_PATH, transactionFile);
+    const fileContent = await services.filesystem.readFile(filePath);
+
+    if (fileContent === null) {
+      throw Error(`Failed to read file at path: ${transactionFilePath}`);
+    }
+    const transaction = JSON.parse(fileContent);
+    if (signed) {
+      tx.ICommandSchema.parse(transaction);
```

Comment:
Parse should also return the data with correct type so you don't need to use "as x"